docs/chap-networking.md

```diff
@@ -214,6 +214,171 @@ The Polkadot Host must actively communicate with the network in order to partici
   The Polkadot network originally only used SCALE encoding for all message formats. Meanwhile, Protobuf has been adopted for certain messages. The encoding of each listed message is always SCALE encoded unless Protobuf is explicitly mentioned. Encoding and message formats are subject to change.
 :::
 
+
+<!-- authority discovery -->
+
+### -sec-num- Discovering authorities {#sect-authority-discovery}
+
+<!-- what it is  -->
+A discovery mechanism enables Polkadot nodes to both publish their local addresses and learn about other nodes identifiers and addresses.
+The Authority discovery mechanism differs from the bootstrap mechanism, described in [Section -sec-num-ref-](chap-networking#sect-network-bootstrap), in that it restricts the discovery output to nodes currently holding the authority role (e.g., validators).
+
+The authority discovery mechanism consists of two main interfaces. A `service` interface that exposes two functions that allow Polkadot nodes to request  `identifiers` or  `addresses` of current authority nodes. A `worker` interface that provides means of interacting with the Kademlia DHT in order to publish local node addresses and discover the addresses of other authority nodes.
+
+```
+       -------
+      |  Node |
+       -------
+          |
+ ---------------------
+| Discovery Authority |
+ ---------------------
+          |
+       -------
+      |  DHT  |
+       -------        
+```
+
+<!-- POLKADOT NODES interacting with AUTHORITY DISCOVERY -->
+#### -sec-num- Requesting authority identifier and addresses {#sect-auth-discovery-service-requests}
+<!-- `GetAuthorityIdsByPeerId` and `GetAddressesByAuthorityId` are the type of message sent from `service` to `worker` in order to receive the information requested by polkadot nodes. However, this is a implementation detail that can be transparent to developers. 
+ -->
+The following requests are exposed by the discovery authority mechanism to Polkadot nodes.
+
+###### Definition -def-num- Authority addresses request {#defn-auth-discovery-address-request}
+:::definition
+
```

Comment:
(I'm commenting on lines 226 to 306, I have some issues with GitHub's UI)

Again this is not supposed to be in a specification.
The fact that Substrate/Polkadot use this service and worker system is completely an implementation detail.
Could an implementation of Polkadot **not** use this service and worker, or could it use different requests, and still work at 100% of its capacity? Yes, completely. Consequently, not in the specification.

A specification is only to define how implementations of the Polkadot communicate with each other and how they must behave.

Reply:
I understand your point, but the text is merely explaining the need for the Authority discovery mechanism to provide two distinguished interfaces and what data is exchanged on each one. The use of names similar to substrate components in the text is purely for illustrative purposes and doesn't imply a direct reference to any specific substrate component. It's simply a choice of terminology to convey the concept.

Reply:
While having a service and a worker in a system can be indicative of using an actor model, it is not a definitive indicator on its own. Having a service and a worker could be part of an actor model-based if the components are implemented as independent actors communicating through message passing. However, having a service and a worker could also be part of other architectural patterns or paradigms, such as a client-server, a microservices, or a task-based system, without necessarily adhering to the actor model. The goal here is not to impose any model.